might need "tougher" has_symlink_capability or some other issue on windows in test_status_symlinked_dir_within_repo

Running tests against fresh build of git-annex (within datalad/git-annex) on github windows env has been failing with 

<details>
<summary>following failure of test_status_symlinked_dir_within_repo</summary> 

```shell
2022-07-15T06:57:08.3610087Z ================================== FAILURES ===================================
2022-07-15T06:57:08.3610843Z ____________________ test_status_symlinked_dir_within_repo ____________________
2022-07-15T06:57:08.3611690Z
2022-07-15T06:57:08.3612131Z path = 'C:\\Users\\runneradmin\\AppData\\Local\\Temp\\datalad_temp_zpuf_aqg'
2022-07-15T06:57:08.3612892Z
2022-07-15T06:57:08.3613135Z     @with_tempfile
2022-07-15T06:57:08.3613809Z     def test_status_symlinked_dir_within_repo(path=None):
2022-07-15T06:57:08.3614339Z         if not has_symlink_capability():
2022-07-15T06:57:08.3614872Z             raise SkipTest("Can't create symlinks")
2022-07-15T06:57:08.3615354Z         # <path>
2022-07-15T06:57:08.3615807Z         # |-- bar -> <path>/foo
2022-07-15T06:57:08.3616193Z         # `-- foo
2022-07-15T06:57:08.3616592Z         #     `-- f
2022-07-15T06:57:08.3617038Z         ds = Dataset(path).create()
2022-07-15T06:57:08.3617498Z         foo = ds.pathobj / "foo"
2022-07-15T06:57:08.3618107Z         foo.mkdir()
2022-07-15T06:57:08.3618535Z         (foo / "f").write_text("content")
2022-07-15T06:57:08.3619048Z         (ds.pathobj / "bar").symlink_to(foo, target_is_directory=True)
2022-07-15T06:57:08.3619475Z         ds.save()
2022-07-15T06:57:08.3619909Z         bar_f = ds.pathobj / "bar" / "f"
2022-07-15T06:57:08.3620319Z
2022-07-15T06:57:08.3620695Z         def call():
2022-07-15T06:57:08.3621119Z             return ds.status(path=[bar_f], annex="availability",
2022-07-15T06:57:08.3621659Z                              on_failure="ignore", result_renderer='disabled')
2022-07-15T06:57:08.3622102Z
2022-07-15T06:57:08.3622503Z         if ds.repo.git_annex_version < "8.20200522" or on_windows:
2022-07-15T06:57:08.3623060Z             # TODO: on windows even with a recent annex -- no CommandError is
2022-07-15T06:57:08.3623554Z             # raised, TODO
2022-07-15T06:57:08.3624167Z >           assert_result_count(call(), 0)
2022-07-15T06:57:08.3624473Z
2022-07-15T06:57:08.3625065Z C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\site-packages\datalad\core\local\tests\test_status.py:303:
2022-07-15T06:57:08.3626498Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
2022-07-15T06:57:08.3627365Z
2022-07-15T06:57:08.3628205Z results = [{'action': 'status', 'command': 'find', 'error_message': 'File unknown to git', 'note': 'not found', ...}]
2022-07-15T06:57:08.3628882Z n = 0, kwargs = {}, count = 1
2022-07-15T06:57:08.3629321Z
2022-07-15T06:57:08.3629739Z     def assert_result_count(results, n, **kwargs):
2022-07-15T06:57:08.3630464Z         """Verify specific number of results (matching criteria, if any)"""
2022-07-15T06:57:08.3631428Z         count = 0
2022-07-15T06:57:08.3632018Z         results = ensure_list(results)
2022-07-15T06:57:08.3632698Z         for r in results:
2022-07-15T06:57:08.3633264Z             if not len(kwargs):
2022-07-15T06:57:08.3633979Z                 count += 1
2022-07-15T06:57:08.3634730Z             elif all(k in r and r[k] == v for k, v in kwargs.items()):
2022-07-15T06:57:08.3635450Z                 count += 1
2022-07-15T06:57:08.3636020Z         if not n == count:
2022-07-15T06:57:08.3637041Z             raise AssertionError(
2022-07-15T06:57:08.3637774Z                 'Got {} instead of {} expected results matching\n{}\nInspected {} record(s):\n{}'.format(
2022-07-15T06:57:08.3638433Z                     count,
2022-07-15T06:57:08.3639021Z                     n,
2022-07-15T06:57:08.3639533Z                     _format_res(kwargs),
2022-07-15T06:57:08.3640128Z                     len(results),
2022-07-15T06:57:08.3640711Z >                   _format_res(results)))
2022-07-15T06:57:08.3641375Z E           AssertionError: Got 1 instead of 0 expected results matching
2022-07-15T06:57:08.3641950Z E             {}
2022-07-15T06:57:08.3642793Z E           Inspected 1 record(s):
2022-07-15T06:57:08.3643359Z E             [
2022-07-15T06:57:08.3643822Z E              {
2022-07-15T06:57:08.3644388Z E               "action": "status",
2022-07-15T06:57:08.3645022Z E               "command": "find",
2022-07-15T06:57:08.3645645Z E               "error_message": "File unknown to git",
2022-07-15T06:57:08.3646390Z E               "note": "not found",
2022-07-15T06:57:08.3647184Z E               "parentds": "C:\\Users\\runneradmin\\AppData\\Local\\Temp\\datalad_temp_zpuf_aqg",
2022-07-15T06:57:08.3648080Z E               "path": "C:\\Users\\runneradmin\\AppData\\Local\\Temp\\datalad_temp_zpuf_aqg\\bar\\f",
2022-07-15T06:57:08.3648789Z E               "refds": "C:\\Users\\runneradmin\\AppData\\Local\\Temp\\datalad_temp_zpuf_aqg",
2022-07-15T06:57:08.3649316Z E               "state": "unknown",
2022-07-15T06:57:08.3649688Z E               "status": "error",
2022-07-15T06:57:08.3650113Z E               "success": false
2022-07-15T06:57:08.3650519Z E              }
2022-07-15T06:57:08.3650845Z E             ]
2022-07-15T06:57:08.3651113Z
2022-07-15T06:57:08.3651544Z C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\site-packages\datalad\tests\utils_pytest.py:1526: AssertionError

```
</details>

which implies that on that windows system `has_symlink_capability` returns True since the test is not skipped.  But then test itself fails.

<details>
<summary>  Interestingly some other tests which seems to deal with symlinks pass
</summary> 

```shell
(git)smaug:…s/2022/07[master]cron-20220715/build-windows.yaml-706-841b46fb-failed/test-datalad (maint)
$> grep link * 
22_Run datalad tests.txt:2022-07-15T04:49:56.7819933Z ..\datalad\core\local\tests\test_save.py::test_symlinked_relpath create(ok): . (dataset)
22_Run datalad tests.txt:2022-07-15T04:53:42.5104225Z add(ok): symlink (file)
22_Run datalad tests.txt:2022-07-15T04:59:12.4245175Z ..\datalad\core\local\tests\test_status.py::test_status_symlinked_dir_within_repo create(ok): . (dataset)
22_Run datalad tests.txt:2022-07-15T04:59:48.7190859Z ..\datalad\customremotes\tests\test_archives.py::test_link_file_load SKIPPED
22_Run datalad tests.txt:2022-07-15T05:02:45.7347779Z drop(impossible): subds_modified\link2superdsdir (symlink) [cannot drop untracked content, save first]
22_Run datalad tests.txt:2022-07-15T05:02:45.8177116Z drop(impossible): link2dir (symlink) [cannot drop untracked content, save first]
22_Run datalad tests.txt:2022-07-15T05:02:45.8179674Z drop(impossible): link2subdsdir (symlink) [cannot drop untracked content, save first]
22_Run datalad tests.txt:2022-07-15T05:02:45.8181907Z drop(impossible): link2subdsroot (symlink) [cannot drop untracked content, save first]
22_Run datalad tests.txt:2022-07-15T05:02:45.8186614Z drop(impossible): subdir\link2annex_files.txt (symlink) [cannot drop untracked content, save first]
22_Run datalad tests.txt:2022-07-15T05:08:52.4107112Z ..\datalad\distribution\tests\test_dataset.py::test_symlinked_dataset_properties SKIPPED
22_Run datalad tests.txt:2022-07-15T05:08:52.4270695Z ..\datalad\distribution\tests\test_dataset.py::test_resolve_path_symlink_edition PASSED
22_Run datalad tests.txt:2022-07-15T05:45:42.2287954Z add(ok): links.json (file)
22_Run datalad tests.txt:2022-07-15T06:00:00.5096796Z add(ok): directory_untracked\link2dir (file)
22_Run datalad tests.txt:2022-07-15T06:00:00.5097949Z add(ok): link2dir (file)
22_Run datalad tests.txt:2022-07-15T06:00:00.5098533Z add(ok): link2subdsdir (file)
22_Run datalad tests.txt:2022-07-15T06:00:00.5098988Z add(ok): link2subdsroot (file)
22_Run datalad tests.txt:2022-07-15T06:00:00.5099399Z add(ok): subdir\link2annex_files.txt (file)
22_Run datalad tests.txt:2022-07-15T06:00:05.2096665Z Removing link2superdsdir
22_Run datalad tests.txt:2022-07-15T06:01:27.9103330Z add(ok): subdir\link2annex_files.txt (file)
22_Run datalad tests.txt:2022-07-15T06:01:31.1594988Z remove(ok): subdir\link2annex_files.txt (file)
22_Run datalad tests.txt:2022-07-15T06:01:46.2106100Z add(ok): directory_untracked\link2dir (file)
22_Run datalad tests.txt:2022-07-15T06:01:46.2106937Z add(ok): link2dir (file)
22_Run datalad tests.txt:2022-07-15T06:01:46.2107565Z add(ok): link2subdsdir (file)
22_Run datalad tests.txt:2022-07-15T06:01:46.2108035Z add(ok): link2subdsroot (file)
22_Run datalad tests.txt:2022-07-15T06:55:34.3073630Z ..\datalad\tests\test_tests_utils_pytest.py::test_ok_symlink_helpers PASSED
22_Run datalad tests.txt:2022-07-15T06:56:04.4829718Z ..\datalad\tests\test_tests_utils_pytest.py::test_ok_file_under_git_symlinks PASSED
22_Run datalad tests.txt:2022-07-15T06:56:06.9505098Z ..\datalad\tests\test_utils.py::test_getpwd_symlink SKIPPED (Skippin...)
22_Run datalad tests.txt:2022-07-15T06:57:08.3610843Z ____________________ test_status_symlinked_dir_within_repo ____________________
22_Run datalad tests.txt:2022-07-15T06:57:08.3613809Z     def test_status_symlinked_dir_within_repo(path=None):
22_Run datalad tests.txt:2022-07-15T06:57:08.3614339Z         if not has_symlink_capability():
22_Run datalad tests.txt:2022-07-15T06:57:08.3614872Z             raise SkipTest("Can't create symlinks")
22_Run datalad tests.txt:2022-07-15T06:57:08.3619048Z         (ds.pathobj / "bar").symlink_to(foo, target_is_directory=True)
22_Run datalad tests.txt:2022-07-15T06:57:08.3660575Z FAILED ..\datalad\core\local\tests\test_status.py::test_status_symlinked_dir_within_repo

```
</details>

and here is our github workflow for windows https://github.com/datalad/git-annex/blob/HEAD/.github/workflows/build-windows.yaml where I do not think we enable any symlinks... From issue like https://github.com/actions/virtual-environments/issues/3065 I guess that may be that windows env is with experimental feature of symlinks on windows turned on!  

Meanwhile I will mark it as xfail if on windows but having symlinks support since this seems to be the only one failing on windows ATM.

--- a/datalad/core/local/tests/test_status.py
+++ b/datalad/core/local/tests/test_status.py
@@ -297,9 +297,11 @@ def test_status_symlinked_dir_within_repo(path=None):
         return ds.status(path=[bar_f], annex="availability",
                          on_failure="ignore", result_renderer='disabled')
 
-    if ds.repo.git_annex_version < "8.20200522" or on_windows:
-        # TODO: on windows even with a recent annex -- no CommandError is
-        # raised, TODO
+    if ds.repo.git_annex_version < "8.20200522" \
+        or (on_windows and ds.repo.git_annex_version < "10.20220525"):
+        # version for windows is an approx guess, but stopped happening
+        # somewhere around 10.20220505-g3b83224e5 may be.
+        # see https://github.com/datalad/datalad/issues/6849
         assert_result_count(call(), 0)
     elif ds.repo.git_annex_version < '10.20220222':
         # As of 2a8fdfc7d (Display a warning message when asked to operate on a

